py3status/modules/imap.py

```diff
@@ -325,19 +360,58 @@ def _get_mail_count(self):
                         auth_string = "user={}\1auth=Bearer {}\1\1".format(
                             self.user, self.creds.token
                         )
-                        self.connection.authenticate("XOAUTH2", lambda x: auth_string)
+                        self.connection.authenticate(
+                            "XOAUTH2", lambda x: auth_string
+                        )
                     else:
                         # Login with user and password
                         self.connection.login(self.user, self.password)
 
                 tmp_mail_count = 0
-                directories = self.mailbox.split(",")
+                if self.mailbox:
+                    directories = self.mailbox.split(",")
+                else:
+                    try:
+                        directories
+                    except:
+                        directories = []
+                        filters = "|".join(self.filter)
+                        objs = [x.decode() for x in self.connection.list()[1]]
+                        folders = [x[-1] for x in reader(objs, delimiter=" ")]
+                        lines = ["===== IMAP ====="]
+                        for name in folders:
+                            subscribed = " "
+                            try:
+                                if re.search(filters, name):
+                                    subscribed = "x"
+                                    folder = name.replace("\\", "\\\\")
+                                    folder = folder.replace('"', '\\"')
+                                    folder = '"{}"'.format(folder)
+                                    directories.append(folder)
+                            except re.error:
+                                directories = []
+                                break
+                            lines.append("[{}] {}".format(subscribed, name))
+                        if not directories:
+                            self.py3.error(
+                                STRING_INVALID_FILTER.format(filters),
+                                self.py3.CACHE_FOREVER,
+                            )
+                        for line in lines:
+                            self.py3.log(line)
```

Comment:
that should be removed, otherwise the log gets spammed constantly. same goes for `lines = ["===== IMAP ====="]`, and everything that's connected to it (e.g. '`subscribed`)

Reply:
it should only be printed once as in line 374,375 the try checks if the variable directorys is defined and if it is not runs the regexes. This is also important as the regex can be quite heavy on cpu. It would be possible to put it behind a debug flag but i would leave it in there to be able to debug the regex.